test/unit/token/colmetadata-token-parser-test.js

```diff
@@ -20,20 +21,19 @@ describe('Colmetadata Token Parser', () => {
     buffer.writeBVarchar(columnName);
     // console.log(buffer.data)
 
-    const parser = new TokenStreamParser({ token() { } }, {}, {});
-    parser.write(buffer.data);
-    const token = parser.read();
-    // console.log(token)
+    const parser = StreamParser.parseTokens([buffer.data], {}, {});
 
-    assert.isOk(!token.error);
-    assert.strictEqual(token.columns.length, 1);
-    assert.strictEqual(token.columns[0].userType, 2);
-    assert.strictEqual(token.columns[0].flags, 3);
-    assert.strictEqual(token.columns[0].type.name, 'Int');
-    assert.strictEqual(token.columns[0].colName, 'name');
+    for await (let token of parser) { // (4)
+      assert.isOk(!token.error);
+      assert.strictEqual(token.columns.length, 1);
+      assert.strictEqual(token.columns[0].userType, 2);
+      assert.strictEqual(token.columns[0].flags, 3);
+      assert.strictEqual(token.columns[0].type.name, 'Int');
+      assert.strictEqual(token.columns[0].colName, 'name');
+    }
```

Comment:
What do you think about changing this to explicit iteration, like this:

```suggestion
    let result = await parser.next();
    assert.isFalse(result.done);

    const token = result.value;
    assert.isOk(!token.error);
    assert.strictEqual(token.columns.length, 1);
    assert.strictEqual(token.columns[0].userType, 2);
    assert.strictEqual(token.columns[0].flags, 3);
    assert.strictEqual(token.columns[0].type.name, 'Int');
    assert.strictEqual(token.columns[0].colName, 'name');

    result = await parser.next();
    assert.isTrue(result.done);
```

I think this is a bit closer to the old test code. It also is more robust, e.g. most of the `for await` based tests would pass if we e.g. introduced a bug that accidentally returned each token twice or more times.